DBZ-2646 Add k8s deployment doc (#3277)
debezium/debezium: documentation/modules/ROOT/pages/operations/kubernates.adoc

[id="deploying-debezium-on-kubernates"]

> = Deploying {prodname} on Kubernates

Reviewer: ```suggestion
= Deploying {prodname} on Kubernetes
```

Same in the file name itself and elsewhere in this file. Best do a search/replace everywhere.

:linkattrs:
:icons: font
:toc:
:toclevels: 3
:toc-placement: macro

toc::[]

> {prodname} can be easily deployed on https://kubernetes.io/[Kubernates], an open-source container management platform.

Reviewer: ```suggestion
{prodname} can be easily deployed on https://kubernetes.io/[Kubernetes], an open-source container management platform.
```

> The deployment leverages https://strimzi.io/[Strimzi] project, which aims to simplify Apache Kafka deployment on Kubernates.

Reviewer: ```suggestion
The deployment leverages the https://strimzi.io/[Strimzi] project, which aims to simplify the deployment of Apache Kafka on Kubernetes by means of _custom resources_.
```

[NOTE]
====

> For testing you deployment, you can use https://minikube.sigs.k8s.io/docs/[minikube] project, which starts Kubernates cluster on your local machine.

Reviewer: ```suggestion
For testing your deployment, you can use https://minikube.sigs.k8s.io/docs/[minikube], which starts a Kubernetes cluster on your local machine.
```

> See https://minikube.sigs.k8s.io/docs/start/[minikube documentation] how to install minikube on your machine.

Reviewer: ```suggestion
See the https://minikube.sigs.k8s.io/docs/start/[minikube documentation] for details on how to install minikube on your machine.
```

> If you want to test {prodname} deployment as described in this doc completely on minikube, you need to set up unsecure Docker registry on minikube.

Reviewer: ```suggestion
If you want to test a {prodname} deployment as described in this doc completely on minikube, you need to set up an unsecure container image registry on minikube.
```

> To do so, you need start minikube with `--insecure-registry` flag:

Reviewer: ```suggestion
To do so, you need start minikube with the `--insecure-registry` flag:
```

[source,subs="attributes",options="nowrap"]
----
minikube start --insecure-registry "10.0.0.0/24"
----

> `10.0.0.1` is the default service cluster IP, so this setup allows pulling images inside whole cluster.

Reviewer: ```suggestion
`10.0.0.1` is the default service cluster IP, so this setup allows pulling images inside the whole cluster.
```
